opentelemetry-otlp/tests/integration_test/README.md

```diff
@@ -15,3 +15,42 @@ popped back out into the files output by the collector.
+| **Simple**           | Inside `tokio::Runtime`  | Outside `tokio::Runtime` | `hyper`                   | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `tonic`                   | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `reqwest`                 | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `reqwest-blocking`        | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `hyper`                   | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `tonic`                  | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `reqwest`                | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `reqwest-blocking`       | Works      |
+| **Simple**           | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `hyper`                  | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Inside `tokio::Runtime` | `tonic`                   | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Inside `tokio::Runtime` | `reqwest`                 | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Inside `tokio::Runtime` | `reqwest-blocking`        | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Inside `tokio::Runtime` | `hyper`                   | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Outside `tokio::Runtime` | `tonic`                  | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Outside `tokio::Runtime` | `reqwest`                | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Outside `tokio::Runtime` | `reqwest-blocking`       | Works      |
+| **Batch**            | Inside `tokio::Runtime`  | Outside `tokio::Runtime` | `hyper`                  | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `tonic`                  | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `reqwest`                | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `reqwest-blocking`       | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Inside `tokio::Runtime` | `hyper`                  | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `tonic`                 | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `reqwest`               | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `reqwest-blocking`      | Works      |
+| **Batch**            | Outside `tokio::Runtime` | Outside `tokio::Runtime` | `hyper`                 | Works      |
+
```

Comment:
This is not really related to the PR at hand, but I also remain unconvinced that we are gaining much by supporting hyper/tonic/reqwest all at once; we started discussing this elsewhere (but I can't find the PR anymore). I struggle to find a strong need for anything more than a single client for async and sync calling, as it is 1/ an implementation detail that does not concern the caller and 2/ likely won't save space in the app anyway because of the way rust manages dep resolution.

Reply:
Similar to async part, Rust does not have a std library for making Http calls or asynchronous runtimes... So we need to provide some ability for users to bring-their-own, and provide a default implementation for at least 1 common one.